game-core/src/test/java/games/strategy/triplea/delegate/battle/steps/sub/OffensiveSubsRetreatTest.java

```diff
@@ -0,0 +1,196 @@
+                .attackingUnits(List.of(givenUnitCanEvade()))
+                .defendingUnits(List.of(givenUnitDestroyer()))
+                .build()),
+        Arguments.of(
+            "Waiting to Die Defending Destroyer exists",
+            givenBattleStateBuilder()
+                .attackingUnits(List.of(givenUnitCanEvade()))
+                .defendingWaitingToDie(List.of(givenUnitDestroyer()))
+                .build()),
+        Arguments.of(
+            "Amphibious assault",
+            givenBattleStateBuilder()
+                .attackingUnits(List.of(givenUnitCanEvade()))
+                .gameData(
+                    MockGameData.givenGameData().withTransportCasualtiesRestricted(false).build())
+                .amphibious(true)
+                .build()),
+        Arguments.of(
+            "Can withdraw but only defenseless transports on the defense",
+            givenBattleStateBuilder()
+                .attackingUnits(List.of(givenUnitCanEvade()))
+                .gameData(
+                    MockGameData.givenGameData()
+                        .withTransportCasualtiesRestricted(true)
+                        .withSubmersibleSubs(false)
+                        .build())
+                .defendingUnits(List.of(givenUnitTransport()))
+                .attackerRetreatTerritories(List.of(mock(Territory.class)))
+                .build()),
+        Arguments.of(
+            "Can submerge but only defenseless transports on the defense",
+            givenBattleStateBuilder()
+                .attackingUnits(List.of(givenUnitCanEvade()))
+                .gameData(
+                    MockGameData.givenGameData()
+                        .withTransportCasualtiesRestricted(true)
+                        .withSubmersibleSubs(true)
+                        .build())
+                .defendingUnits(List.of(givenUnitTransport()))
+                .build()),
+        Arguments.of(
+            "No retreat territories and can not submerge",
+            givenBattleStateBuilder()
+                .gameData(
+                    MockGameData.givenGameData().withTransportCasualtiesRestricted(false).build())
+                .attackerRetreatTerritories(List.of())
+                .build()));
+  }
+
+  static class MockGameData {
+    private final GameData gameData;
+    private final GameProperties gameProperties;
+    private boolean propertiesSetup = false;
+
+    private MockGameData() {
+      gameData = mock(GameData.class);
+      gameProperties = mock(GameProperties.class);
+    }
+
+    static MockGameData givenGameData() {
+      return new MockGameData();
+    }
+
+    GameData build() {
+      return gameData;
+    }
+
+    MockGameData withTransportCasualtiesRestricted(final boolean value) {
+      setupProperties();
+      when(gameProperties.get(TRANSPORT_CASUALTIES_RESTRICTED, false)).thenReturn(value);
+      return this;
+    }
+
+    private void setupProperties() {
+      if (!propertiesSetup) {
```

Comment:
Is this if check here to avoid strict mocking checks?
It might be a case where we can use lenient. IMO using lenient is going to be better than having lazy instantiation of a mock expectation.

Reply:
Yes, it is to avoid strict mocking checks.

Reply:
Do you have objection to using a lenient? It seems like a pretty good case for it. Avoiding state in test scaffolding is generally a very good thing.

Reply:
I'm fine with lenient.